float32 x
float32 y
float32 v
float32 a
float32 theta
float32 theta_rate